packages/openauth/src/storage/sqlite.ts

```diff
@@ -0,0 +1,59 @@
+import { joinKey, splitKey, type StorageAdapter } from "./storage.js";
+import Database from "libsql";
+
+export interface SqliteStorageOptions {
+  persist?: string;
+  tableName?: string;
+}
+
+export function SQLiteStorage(input?: SqliteStorageOptions): StorageAdapter {
+  // initialize sqlite database and create the necessary table structure
+  const db = new Database(input?.persist ?? ":memory:");
+  const TABLE_NAME = input?.tableName ?? "__openauth__kv_storage";
+
+  db.exec(
+    `CREATE TABLE IF NOT EXISTS ${TABLE_NAME} (key TEXT PRIMARY KEY, value TEXT, expiry INTEGER)`
+  );
+
+  return {
+    async get(key: string[]) {
+      const joined = joinKey(key);
+
+      const row = db
+        .prepare(`SELECT value, expiry FROM ${TABLE_NAME} WHERE key = ?`)
+        .get(joined) as { value: string; expiry: number } | undefined;
+
+      if (row && row.expiry && row.expiry < Date.now()) {
+        db.prepare(`DELETE FROM ${TABLE_NAME} WHERE key = ?`).run(joined);
+        return undefined;
+      }
+      return row ? JSON.parse(row.value) : undefined;
+    },
+    async set(key: string[], value: any, ttl?: number) {
+      const expiry = ttl ? Date.now() + ttl * 1000 : undefined;
```

Comment:
This is already set in https://github.com/openauthjs/openauth/blob/master/packages/openauth/src/storage/storage.ts#L33

```suggestion
    async set(key: string[], value: any, expiry?: Date) {
```